### PR TITLE
Add dependency counting tool

### PR DESCRIPTION
**Pull request by cif**
Useful for identifying "weak" links in the graph where there is less work moving things around which can still reduce the number of edges and total dependencies of the main app/functions programs.

`bun src/index.ts count-dependency-imports ../glide`

produces output like this: 

```
Analyzing @glide/billing-entitlements-tool, contains 7 dependencies total, counting imports for each...
Used dependencies and their counts:
0: @glide/billing-types,9
1: @glide/billing-entitlements-svc,6
2: @glide/dba-utils,5
3: @glide/support,3
4: @glide/honeycomb,2
5: @glide/observability,2
6: @glide/config,1


Analyzing @glide/billing-entitlements-svc, contains 10 dependencies total, counting imports for each...
Used dependencies and their counts:
0: @glide/billing-types,8
1: @glide/dba-utils,5
2: @glide/indexeddb,2
3: @glide/plugins,2
4: @glide/plugins-utils,2
5: @glide/support,1
6: @glide/glide-plugins,1
7: @glide/config,1
8: @glide/honeycomb,1
9: @glide/observability,1

etc.
```

### CODE REVIEW

File: src/count-dependency-imports.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    if (pkg === "@glide/app") {
+      actualDir = path.join(dir, "app");
+    }
+
+    if (pkg === "@glide/functions") {
+      actualDir = path.join(dir, "functions");
+    }
+
+    const tsFiles = glob.sync(path.join(actualDir, "**/*.ts*"));
```
Comment (schani) [inline src/count-dependency-imports.ts]: Bun has a new/better/faster glob API now, btw: https://bun.sh/blog/bun-v1.1#bun-glob

File: src/count-dependency-imports.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    // Use a Map to store and count dependencies
+    const dependencyCounts = new Map();
+
+    // Updated regex to capture the full import path
+    const importExportRegex = /from\s+['"](@?[^'"]+)['"]/g;
```
Comment (schani) [inline src/count-dependency-imports.ts:64]: I recommend using `parseImports`

**Review by schani — APPROVED** [review]
Pull request 1 - amazing!!!!

No changes necessary, just FYI.

File: src/count-dependency-imports.ts (hunk 2 above)
Comment (cif) [reply src/count-dependency-imports.ts:64]: Awesome will do. I don't think this regex courtesy of chat GPT is quite accurate, but it was good enough in my initial quest to find some easy wins

Comment (schani) [reply src/count-dependency-imports.ts:64]: We also have `walkDirectory`, which in hindsight should actually just use globbing, not walking.

**Pull request merged**